### PR TITLE
using semaphore in tpu connection manager and removing unwanted code

### PR DESCRIPTION
**Pull request by godmodegalactus**

### CODE REVIEW

File: lite-rpc/src/lib.rs [hunk 1]
```diff
@@ -28,7 +28,7 @@ pub const DEFAULT_FANOUT_SIZE: u64 = 10;
 #[from_env]
 pub const MAX_RETRIES: usize = 40;
 
-pub const DEFAULT_RETRY_TIMEOUT: u64 = 2;
+pub const DEFAULT_RETRY_TIMEOUT: u64 = 1;
```
Comment (grooviegermanikus) [inline lite-rpc/src/lib.rs:31]: can this be _Duration_ typed or have at least a suffix and/or comment if it's seconds/milliseconds/etc.

File: services/src/tpu_utils/tpu_connection_manager.rs [hunk 2]
```diff
@@ -156,13 +151,12 @@ impl ActiveConnection {
 
                     let task_counter = task_counter.clone();
                     let connection_pool = connection_pool.clone();
-
+                    let permit = task_counter.acquire_owned().await.expect("Should get permit");
```
Comment (grooviegermanikus) [inline services/src/tpu_utils/tpu_connection_manager.rs:154]: could you put a comment, what the expected behavior should be wrt backpressure? more transactions will be .recv'd and block the select-handler?
wouldn't it be better to keep the transactions in the channel?

File: lite-rpc/src/lib.rs (hunk 1 above)
Comment (godmodegalactus) [reply lite-rpc/src/lib.rs:31]: Duration is used in CLI as a default and Duration is incompatible with Clap

Comment (grooviegermanikus) [reply lite-rpc/src/lib.rs:31]: k. got it, that's sad

**Review by grooviegermanikus — APPROVED** [review]
k

**Pull request merged**